clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3002,6 +3019,12 @@ static void handleNumSimdWorkItemsAttr(Sema &S, Decl *D,
   if (D->getAttr<SYCLIntelNumSimdWorkItemsAttr>())
     S.Diag(Attr.getLoc(), diag::warn_duplicate_attribute) << Attr;
 
+  if (checkDeprecatedFPGASYCLSpelling(S, Attr)) {
+    S.Diag(Attr.getLoc(), diag::note_spelling_suggestion)
+        << "'intel::num_simd_work_items'";
+    return;
```

Comment:
So, if an attribute has deprecated spelling we emit a warning and don't add this attribute to a declaration, right?

Reply:
@Fznamznon yes, correct.

Reply:
We should probably add the attribute to the declaration still.  "Deprecated" typically means "This still works, but we're going to make it stop working in the future".

Reply:
> We should probably add the attribute to the declaration still. "Deprecated" typically means "This still works, but we're going to make it stop working in the future".

Yep, that is what I'd expect from "deprecated" attribute.

Reply:
Agree that we shall have the attribute added to the decl.

Reply:
I have updated patch to fix the issue. Now this adds attribute to the declaration.  Deprecated attribute spelling works same as new spelling.